container-storage-setup.sh

```diff
@@ -1385,6 +1402,29 @@ setup_storage_compat() {
     Error "Failed to setup logical volume for $CONTAINER_ROOT_LV_MOUNT_PATH."
     return 1
   fi
+
+  # For case of overlay/ovelray2, make sure ftype is enabled.
+  if [ -n "$_RESOLVED_MOUNT_DIR_PATH" ]; then
+    containerroot="$_RESOLVED_MOUNT_DIR_PATH"
+  else
+    containerroot="/var"
+  fi
+
+  if [ "$STORAGE_DRIVER" == "overlay" -o "$STORAGE_DRIVER" == "overlay2" ]; then
+    # This is little hacky. We are guessing where overlay2 will be setup
+    # by container runtime environment. At some point of time this should
+    # be passed in by a config variable.
+    if [ -n "$_RESOLVED_MOUNT_DIR_PATH" ]; then
+      containerroot="$_RESOLVED_MOUNT_DIR_PATH"
+    else
+      containerroot="/var"
+    fi
```

Comment:
`containerroot=${_RESOLVED_MOUNT_DIR_PATH:-/var}` ?

Reply:
Sounds good. I am not good at using more concise programming constructs. :-) Will change it.